actor/builtin/miner/miner.go

```diff
@@ -809,11 +809,11 @@ func (ma *Actor) GetProvingPeriodStart(ctx exec.VMContext) (*types.BlockHeight,
 		return nil, errors.CodeError(err), err
 	}
 
-	return state.ProvingPeriodStart, 0, nil
+	return state.ProvingPeriodEnd, 0, nil
 }
 
 func currentProvingPeriodPoStChallengeSeed(ctx exec.VMContext, state State) (types.PoStChallengeSeed, error) {
-	bytes, err := ctx.SampleChainRandomness(state.ProvingPeriodStart)
+	bytes, err := ctx.SampleChainRandomness(state.ProvingPeriodEnd)
```

Comment:
BLOCKING: Suppose a miner completes the PoSt generation routine some number of blocks N before the end of their proving period. If N is greater than `sampling.LookbackParameter`, then there will exist no block with a matching height from which to sample randomness from.